src/pay/payment_producer.py

```diff
@@ -98,94 +99,86 @@ def retry_fail_run(self):
         pass
 
     def run(self):
-        # call first retry if not in onetime mode.
-        # retry_failed script is more suitable for one time cases.
-        if not self.run_mode == RunMode.ONETIME:
-            self.retry_failed_payments()
-
-        # first retry is done by producer thread, start retry thread for further retries
-        if self.run_mode == RunMode.FOREVER:
-            self.retry_fail_thread.start()
-
-        current_cycle = self.block_api.get_current_cycle()
-        pymnt_cycle = self.initial_payment_cycle
-
-        # if non-positive initial_payment_cycle, set initial_payment_cycle to
-        # 'current cycle - abs(initial_cycle) - (NB_FREEZE_CYCLE+1)'
-        if self.initial_payment_cycle <= 0:
-            pymnt_cycle = current_cycle - abs(self.initial_payment_cycle) - (self.nw_config['NB_FREEZE_CYCLE'] + 1)
-            logger.debug("Payment cycle is set to {}".format(pymnt_cycle))
-
-        while not self.exiting and self.life_cycle.is_running():
-
-            # take a breath
-            time.sleep(5)
+        if self.run_mode == RunMode.RETRY_FAILED:
+            self.retry_failed_payments(self.retry_injected)
+            time.sleep(10)
```

Comment:
That was my first implementation. But if you return here, the main function would run forever still the life cycle is running. The main thread is interrupted properly (=> set right flags and delete lock file) using the exit method. However, alternative suggestions to fix this would be welcome!

Reply:
Thanks for the suggestions. I implemented the recommended changes. Since we are now interrupting the main thread directly, the interrupt is visible in the log as shown in the image. I am not sure how this can be avoided by avoiding changes to payment_producer.py
![image](https://user-images.githubusercontent.com/33556352/93667646-db993000-fa87-11ea-9bf2-4919d6b64ce7.png)
I personally think that 635942d is more stable even if it includes changes to payment_producer.py.